### PR TITLE
[enhancement](Nereids)enable explain query for nereids planner

### PR DESCRIPTION
**Pull request by morrySnow**
## Problem Summary:

Support explain syntax. And generate explain tree string by nereids' planner.

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: Yes
3. Has document been added or modified: No Need
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

### CODE REVIEW

**Comment by morrySnow** [comment]
@924060929 PTAL

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/NereidsParser.java [hunk 1]
```diff
@@ -46,14 +49,25 @@ public class NereidsParser {
...
-        List<StatementBase> statementBaseList = new ArrayList<>();
-        for (LogicalPlan logicalPlan : logicalPlanList) {
+    public List<StatementBase> parseSQL(String originStr) {
+        List<LogicalPlan> logicalPlans = parseMultiple(originStr);
+        List<StatementBase> statementBases = new ArrayList<>();
+        for (LogicalPlan logicalPlan : logicalPlans) {
+            // TODO: this is a trick to support explain. Since we do not support any other command in a short time.
+            //     It is acceptable. In the future, we need to refactor this.
+            ExplainOptions explainOptions = null;
+            if (logicalPlan instanceof ExplainCommand) {
```
Comment (wangshuo128) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/NereidsParser.java:59]: I think it's clear like this:
```java
            if (logicalPlan instanceof ExplainCommand) {
                ExplainCommand explainCommand = (ExplainCommand) logicalPlan;
                LogicalPlan innerPlan = explainCommand.getLogicalPlan();
                LogicalPlanAdapter logicalPlanAdapter = new LogicalPlanAdapter(innerPlan);
                logicalPlanAdapter.setIsExplain(new ExplainOptions(
                        explainCommand.getLevel() == ExplainLevel.VERBOSE,
                        explainCommand.getLevel() == ExplainLevel.GRAPH));
                statementBases.add(logicalPlanAdapter);
            } else {
                statementBases.add(new LogicalPlanAdapter(logicalPlan));
            }
```

**Review by wangshuo128 — DISMISSED** [review]
LGTM

**Review by 924060929 — DISMISSED** [review]
LGTM

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**